### PR TITLE
Collection trigger fix

### PR DESCRIPTION
**Pull request by shabanovd**

### CODE REVIEW

File: src/org/exist/collections/CollectionCache.java [hunk 1]
```diff
@@ -55,6 +55,9 @@ public void add(Collection collection) {
     }
 
     public void add(Collection collection, int initialRefCount) {
+        // don't cache the collection during initialization: SecurityManager is not yet online
+        if(pool.isInitializing()) return;
```
Comment (dizzzz) [inline src/org/exist/collections/CollectionCache.java]: should there be a log/debug line?

File: src/org/exist/collections/triggers/XQueryTrigger.java [hunk 2]
```diff
@@ -731,16 +707,16 @@ public void beforeCopyDocument(DBBroker broker, Txn txn, DocumentImpl document,
...
 			prepare(4, broker, txn, document.getURI(), newUri, false);
 		} else {
-		    execute(true, broker, txn, beforeCopyDocument, document.getURI());
+            execute(true, broker, txn, beforeCopyDocument, document.getURI(), newUri);
 	    }
 	}
 
 	@Override
-	public void afterCopyDocument(DBBroker broker, Txn txn, DocumentImpl document, XmldbURI newUri) throws TriggerException {
+    public void afterCopyDocument(DBBroker broker, Txn txn, DocumentImpl document, XmldbURI oldUri) throws TriggerException {
```
Comment (dizzzz) [inline src/org/exist/collections/triggers/XQueryTrigger.java]: so the name of the parameter has been wrong, always? Or do I need to update my plugins that rely on the interface?

**Comment by dizzzz** [comment]
good to see a test, most of the code looks straight forward;
I cannot judge on the 'actual' changes though. @adamretter @wolfgangmm @ljo ?

File: src/org/exist/collections/CollectionCache.java (hunk 1 above)
Comment (shabanovd) [inline src/org/exist/collections/CollectionCache.java]: it wasn't before, why now?

File: src/org/exist/collections/triggers/XQueryTrigger.java (hunk 2 above)
Comment (shabanovd) [inline src/org/exist/collections/triggers/XQueryTrigger.java]: if you implement base on interface then names is same, this implementation had typos

**Comment by ChiSamurai** [comment]
Thanks again for your work. Really nice to finally see the triggers running as intended. This fixes https://github.com/eXist-db/exist/issues/255

**Pull request merged**